Makefile

```diff
@@ -34,7 +34,7 @@ help: ## Display this help.
 
 .PHONY: manifests
 manifests: controller-gen ## Generate WebhookConfiguration, ClusterRole and CustomResourceDefinition objects.
-	$(CONTROLLER_GEN) rbac:roleName=odh-model-controller-role,headerFile="hack/manifests_boilerplate.yaml.txt" crd webhook paths="./..." output:crd:artifacts:config=config/crd/bases
```

Comment:
why is `webhook` removed here, but added elsewhere (e.g. `config/default/kustomization.yaml`)?

Reply:
I found that this generator would reset the `config/webhook/manifests.yaml` file to the bare template bundled in controller-gen, which seems to be quite partial and not final.

Since the `manifests` target is also ran in other targets (like `test`), it would be resetting the manifest constantly. So, I thought it was OK to remove it and run the webhook generator on demand when changing/adding webhooks.

Reply:
That's expected, no?
Webhook are configured with annotations in the source code (// comments for kubebuilder)

Reply:
Yes, but the annotations don't let you customize everything.
* There is no way to customize the `metadata.name` that I changed to `validating.odh-model-controller.opendatahub.io`.
* It would remove the `annotations.service.beta.openshift.io/inject-cabundle` for OpenShift serving certificates.

Those are the two most important changes that I did.

So, I think the generator just gives you a YAML as a starting point. It even didn't generate the [required Service](https://github.com/opendatahub-io/odh-model-controller/pull/181/files#diff-4f466ecfd8295868474a2d73456f805b20b06d59ce1b5649777ee1c2371b9085R1), nor did the [changes to the manager](https://github.com/opendatahub-io/odh-model-controller/pull/181/files#diff-ce0d597a9e5f20a8fe61c3ba5e6185a6b90cfef2c00bb5b5ef3e97b15b79f480R77) that are required for the webhook to actually work.

Reply:
Can't you ensure this is defined in `kustomization.yaml` though? Or is it all overwritten?

Reply:
> I thought it was OK to remove it and run the webhook generator on demand when changing/adding webhooks.

That's based on the assumption that one would always remember to do so :)

Reply:
> > I thought it was OK to remove it and run the webhook generator on demand when changing/adding webhooks.
> 
> That's based on the assumption that one would always remember to do so :)

I guess it is OK to forget. If you add a new webhook, it would not work if you forget, so I would expect you would notice it is not working.

> Can't you ensure this is defined in `kustomization.yaml` though? Or is it all overwritten?
You mean to move the edits to the `kustomization.yaml` as patches?
I think I can do that, as long as I can successfully re-name it - I don't recall seeing a patch that would do renaming, as you typically use the name as a selector for patching.

Although, I'm not sure what's the best practice here:
* On KServe repo, [their webhooks](https://github.com/kserve/kserve/blob/master/config/webhook/manifests.yaml) are customized, and they also have removed the generator from their Makefile.
* On opendatahub-operator, they do keep [the autogenerated manifest](https://github.com/opendatahub-io/opendatahub-operator/blob/06256fc6bb2dfbd98e42ea95b0885598d14e2278/config/webhook/manifests.yaml), and they [do add annotations in their kustomization.yaml file](https://github.com/opendatahub-io/opendatahub-operator/blob/06256fc6bb2dfbd98e42ea95b0885598d14e2278/config/webhook/kustomization.yaml#L5-L6). With this they can keep the generator. However, one important point is that odh-operator have preserved [the usage of the namePrefix](https://github.com/opendatahub-io/opendatahub-operator/blob/06256fc6bb2dfbd98e42ea95b0885598d14e2278/config/default/kustomization.yaml#L7-L12) in the entrypoing kustomization.yaml, while here in odh-model-controller, at some point it was chosen to stop using it and that forces us to manage our names. 

I'd appreciate any guidance here.

Reply:
if it is removing already done stuff, maybe it better to keep this way, but create a issue to keep track of it and try to rely on kubebuilder to manage the webhooks.

Reply:
https://issues.redhat.com/browse/RHOAIENG-4752